docs/project/versioning.md

```diff
@@ -0,0 +1,269 @@
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/4105)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Overview](#overview)
+-   [Major version increments](#major-version-increments)
+    -   [Breaking changes](#breaking-changes)
+    -   [Exclusions to what constitutes a breaking change](#exclusions-to-what-constitutes-a-breaking-change)
+-   [Minor version increments](#minor-version-increments)
+-   [Patch version increments](#patch-version-increments)
+    -   [Examples:](#examples)
+-   [Pre-release versions](#pre-release-versions)
+    -   [Release qualification pre-releases: `MAJOR.MINOR.PATCH-rc.N`](#release-qualification-pre-releases-majorminorpatch-rcn)
+    -   [Incremental development versions: `MAJOR.MINOR.PATCH-0.{nightly,dev}.N`](#incremental-development-versions-majorminorpatch-0nightlydevn)
+        -   [Nightly pre-release versions](#nightly-pre-release-versions)
+        -   [Development pre-release versions](#development-pre-release-versions)
+-   [Relevant proposal](#relevant-proposal)
+
+<!-- tocstop -->
+
+## Overview
+
+Carbon uses a single versioning scheme across both the language itself and
+toolchain, including the standard library, compiler, linker, and all development
+tools released by the main Carbon project.
+
+The scheme conforms to and is closely based on Semantic Versioning (SemVer):
+
+-   Carbon versions: `MAJOR.MINOR.PATCH`
+-   `MAJOR` increments on backwards incompatible changes, including a
+    deprecation that might trigger a build-breaking warning.
+-   `MINOR` increments only expected during early development with a major
+    version of `0`.
+-   `PATCH` increments represent bug fixes only.
+-   Pre-release suffixes:
+    -   `MAJOR.MINOR.PATCH-rc.N`: The N-th potentially viable candidate for a
+        release.
+    -   `MAJOR.MINOR.PATCH-0.nightly.YYYY.MM.DD`: A nightly incremental
+        development build on a particular day during development of that
+        version.
+    -   `MAJOR.MINOR.PATCH-0.dev`: An interactive, incremental development build
+        on a particular day by some developer during development of that
+        version.
+
+See the sections below for the details of each aspect of these versions.
+
+## Major version increments
+
+Aligned with SemVer, the major version must increment for any _breaking change_
+to the language or any part of the toolchain.
+
+The first increment from 0 to 1 is expected to be based on achieving a desired
+milestone of feature completeness and quality to declare the language to have
+reached a stable version.
+
+Subsequent increments are expected to be done with a time-based release
+strategy. Features (or breaking changes) ready to ship will do so, and others
+will wait for the next major release. The exact cadence used is future work and
+should be determined based on discussions with customers in the ramp up to and
```

Comment:
Is there a better word than "customers" here, that doesn't suggest a financial relationship? Something like "stakeholders" or "affected parties" perhaps?

I'm worried that people might interpret this as our valuing the views of people who give us money over the rest of the community.

Reply:
Yeah, "users" is a more general and better term.